File: Design Patterns/Singleton Pattern/singleton.py
# This code implements the Singleton pattern in Python.
# It ensures that only one instance of the Singleton class is created and used throughout the program.

class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance

    def some_method(self):
        print("Method of the singleton")

# Usage
instance1 = Singleton()
instance2 = Singleton()

print(instance1 is instance2)  # Output: True

instance1.some_method()  # Output: Method of the singleton

File: Design Patterns/Singleton Pattern/singleton.ts
// This below code show the Singleton pattern in TypeScript.
// It ensures only one instance of the Singleton class can be created and accessed globally.

class Singleton {
  private static instance: Singleton;
  private constructor() {}

  public static getInstance(): Singleton {
    if (!Singleton.instance) {
      Singleton.instance = new Singleton();
    }
    return Singleton.instance;
  }

  public someMethod(): void {
    console.log("Method of the singleton");
  }
}

// Usage
const instance1 = Singleton.getInstance();
const instance2 = Singleton.getInstance();

console.log(instance1 === instance2); // Output: true

instance1.someMethod(); // Output: Method of the singleton
